traits/has_traits.py

```diff
@@ -2333,8 +2334,9 @@ def configure_traits(
         modified by the user.
         """
         if filename is not None:
-            with open(filename, "rb") as fd:
-                self.copy_traits(sm.cPickle.Unpickler(fd).load())
+            if os.path.exists(filename):
+                with open(filename, "rb") as fd:
+                    self.copy_traits(six.moves.cPickle.Unpickler(fd).load())
```

Comment:
do we want to restore the original behavior where we have a try/except around `copy_traits`?

Reply:
Good point. But no, an exception being raised at that point is likely a coding mistake, so let's allow it to propagate.

Reply:
Is it worth it to test/check that the exception is raised?

Reply:
Perhaps. But I'm not sure I see the value in such a test, except as a regression test, and even then I wouldn't expect new code to go back to suppressing all exceptions.

Reply:
I'll add a test with a file whose contents aren't a valid pickle. That'll cover the behaviour change from ancient versions where the resulting exception would have been swallowed.